packages/fx-core/src/common/local/projectSettingsHelper.ts

```diff
@@ -12,20 +17,24 @@ export class ProjectSettingsHelper {
       (pluginName) => pluginName === ResourcePlugins.SPFx
     );
 
-  public static includeFrontend = (projectSettings: ProjectSettings | undefined): boolean =>
-    !!(projectSettings?.solutionSettings as AzureSolutionSettings)?.activeResourcePlugins?.some(
-      (pluginName) => pluginName === ResourcePlugins.FrontendHosting
-    );
-
+  public static includeFrontend = (projectSettings: ProjectSettings | undefined): boolean => {
+    if (!projectSettings) return false;
+    const solutionSettings = projectSettings.solutionSettings as AzureSolutionSettings;
+    return solutionSettings.capabilities.includes(TabOptionItem.id);
```

Comment:
Consider backward compatibility - what if opening existing (before v3) project?
Is `solutionSettings.version` designed for determining whether to use `solutionSettings.activeResourcePlugins` or `solutionSettings.capabilities`?

Reply:
Yes, but a better way is to use capabilities for both versions. Capabilities are the single truth.